### DISCUSSION TITLE
Overriding model property with {% do %} tag not working in Craft 4

### DISCUSSION THREAD

**Discussion by janhenckens** [craftcms/cms#13945] [Q&A] [answered]
In Craft 3, my SEO plugin could do this:

```twig
{% do entry.seo.setTwitterTitle('Twitter title') %}
{% extends '_layout.twig' %}

{% block content %}
    {{ dump(entry.seo.twitterTitle) }} {# Twitter title #}
{% endblock %}
```

But that doesn't work in Craft 4. I've looked at this a bunch of times (as Craft 4 has been out a long while) but I can't seem to get to the bottom of this.

Could something in Craft have changed that prevents this way of working?

**Comment by brandonkelly** [comment, marked as answer]
Nothing in Craft 4 specifically would have affected this. Maybe a breaking change in whatever plugin was providing that `seo.setTwitterTitle()` method?

**Reply by janhenckens** [reply]
So the plugin is my own.
Just tested with a clean Craft 3 install (3.9.6 and the last C3 version of the plugin), and that works.
Upgrading to 4.0.x and the corresponding plugin version breaks it.

I diffed the C3 version versus the C4 version of the plugin and the biggest difference is the change from `Twig_Extension` to `AbstractExtension`. Could the upgrade to Twig 3 and how it handles context cause this?

**Reply by brandonkelly** [reply]
Oh… I’m guessing this is related to #8781. With that change, each time you cal `entry.seo` you are going to get a new clone of the field value.

If you were to do this instead, it should work as expected:

```twig
{% set seo = entry.seo %}
{% do seo.setTwitterTitle('Twitter title') %}
{% extends '_layout.twig' %}

{% block content %}
    {{ dump(seo.twitterTitle) }} {# Twitter title #}
{% endblock %}
```

**Reply by janhenckens** [reply]
Ha, that makes sense actually. With your example I get the custom value in the content block, but not through my template hook that gets called in the layout. Probably because I'm fetching the field from the element again there,  which gets me a new clone.

**Reply by brandonkelly** [reply]
Yeah. You may need to rethink where that value is stored, e.g. on a behavior you attach to the entry (assuming each entry is going to be limited to a single set of SEO values, rather than per-field values.)
